debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/wal2json/Wal2JsonReplicationMessage.java

```diff
@@ -139,105 +149,208 @@ public String getName() {
                 }
-            case "_int2":
-            case "_int4":
-            case "_int8":
-            case "_text":
-            case "_numeric":
-            case "_float4":
-            case "_float8":
-            case "_bool":
-            case "_date":
-            case "_time":
-            case "_timetz":
-            case "_timestamp":
-            case "_timestamptz":
-            case "_bytea":
-            case "_varchar":
-            case "_oid":
-            case "_bpchar":
-            case "_money":
-            case "_name":
-            case "_interval":
-            case "_char":
-            case "_varbit":
-            case "_uuid":
-            case "_xml":
-            case "_point":
-            case "_jsonb":
-            case "_json":
-            case "_ref_cursor":
+            case "point":
                 try {
-                    final String dataString = rawValue.asString();
-                    PgArray arrayData = new PgArray(connection.get(), Oid.valueOf(columnType.substring(1) + "_array"), dataString);
-                    Object deserializedArray = arrayData.getArray();
-                    return Arrays.asList((Object[])deserializedArray);
+                    return new PGpoint(rawValue.asString());
+                } catch (final SQLException e) {
+                    LOGGER.error("Failed to parse point {}, {}", rawValue.asString(), e);
+                    throw new ConnectException(e);
                 }
-                catch (SQLException e) {
-                    LOGGER.warn("Unexpected exception trying to process PgArray column '{}'", columnName, e);
+            case "polygon":
+                try {
+                    return new PGpolygon(rawValue.asString());
+                } catch (final SQLException e) {
+                    LOGGER.error("Failed to parse point {}, {}", rawValue.asString(), e);
+                    throw new ConnectException(e);
                 }
-                return null;
+
+            // catch-all for other known/builtin PG types
+            // TODO: improve with more specific/useful classes here?
+            case "bit":
+            case "bit varying":
+            case "varbit":
+            case "cidr":
+            case "inet":
+            case "json":
+            case "jsonb":
+            case "macaddr":
+            case "macaddr8":
+            case "pg_lsn":
+            case "tsquery":
+            case "tstzrange":
+            case "tsvector":
+            case "txid_snapshot":
+            case "uuid":
+            case "xml":
+                return rawValue.asString();
+
+            default:
+                break;
         }
+        // this includes things like PostGIS geometries or other custom types.
+        // leave up to the downstream message recipient to deal with.
         LOGGER.warn("processing column '{}' with unknown data type '{}' as byte array", columnName,
                 columnType);
         return rawValue.asBytes();
```

Comment:
this path isn't covered by the tests. How to add?

Reply:
a PostGIS geometry column should do it
eg. 
```
CREATE TABLE foo (id int primary key, geom geometry(POINT,4326))
INSERT INTO foo VALUES (1, 'SRID=4326;POINT(174 -36)'::geometry);
```

Reply:
Here are gemotry types tested - https://github.com/debezium/debezium/blob/master/debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/RecordsStreamProducerIT.java#L104-L106

Reply:
hmm, those are Postgresql's native "geometry" types, not the Postgis ones.

Reply:
Yes, but you could extend the table with one of postgis types and check how it is processed